proctor-common/src/main/java/com/indeed/proctor/common/AbstractProctorLoader.java

```diff
@@ -57,6 +58,7 @@ public AbstractProctorLoader(
             @Nonnull final FunctionMapper functionMapper
     ) {
         this(cls, specification, functionMapper, new IdentifierValidator.Noop());
+        loggedDynamicTests = new HashSet<>();
```

Comment:
Should we limit the size of the set so it will not potentially consume too much memory?

Reply:
Checked proctor pipet logs and currently I am seeing 2743 occurrences of this log message per load. What would a good upper bound be?

Reply:
As you explained earlier that loggedDynamicTests only contains test name, so it should be pretty small. A size limit can be optional.